src/goto-programs/remove_exceptions.cpp

```diff
@@ -115,12 +120,25 @@ void remove_exceptionst::add_exceptional_returns(
   // function calls that may escape exceptions. However, this will
   // require multiple passes.
   bool add_exceptional_var=false;
+  bool has_uncaught_exceptions=false;
   forall_goto_program_instructions(instr_it, goto_program)
-    if(instr_it->is_throw() || instr_it->is_function_call())
+  {
+    if(instr_it->is_function_call())
+    {
+      const exprt &function_expr=
+        to_code_function_call(instr_it->code).function();
+      assert(function_expr.id()==ID_symbol);
+      const irep_idt &function_name=
+        to_symbol_expr(function_expr).get_identifier();
+      has_uncaught_exceptions=!exceptions_map[function_name].empty();
```

Comment:
No, as each iteration of the loop is actually independent -- I break as soon as I find a throw or function call that may escape an exception. The scope of ```has_uncaught_exceptions``` was however deceiving so I've reduced it to make the intent clearer.